### DISCUSSION TITLE
How to make aw-server-rust on linux not be in "test" mode?

### DISCUSSION THREAD

**Discussion by pmolodo** [ActivityWatch/activitywatch#758] [Q&A] [answered]
Hi - I tried switching to aw-server-rust, by following these directions:

https://docs.activitywatch.net/en/latest/migrating.html#migrating-to-aw-server-rust

Unfortunately, it didn't work, as it seems aw-server-rust is listening on port 5666, and everything else expects 5600.  Apparently this means it's in "test" mode, but I didn't do anything special to put it in test mode - it's a standard unzipped install on Linux (Ubuntu 20.04, to be precise).

How do I tell aw-server-rust to be in "normal" / non-test mode, and listen on 5600?

Also - how do I turn on debug logging on aw-server-rust?  I tried doing `export RUST_LOG=trace` before launching aw-server-rust, but that didn't seem to have an effect...

**Comment by ErikBjare** [comment, marked as answer]
This is probably a packaging issue. The aw-server-rust binary produced in CI is built in debug mode for some reason (which leads to running in testing mode by default -> port 5666).

I'll look into that.

Edit: Hmm, maybe not. Still investigating.

Edit: Ok nvm, it definitely built in debug mode: https://github.com/ActivityWatch/activitywatch/runs/6724471272

```
 warning: `aw-server` (bin "aw-server") generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 3m 46s
```

Now I just need to figure out why.


Edit: It's because of this https://github.com/ActivityWatch/activitywatch/blob/f5e1580d29333980612d51746d51ae399672ddd0/.github/workflows/build.yml#L17

I need to figure out a way to set that correctly, depending on if it's a proper release or not.

**Reply by pmolodo** [reply]
Ah, well that explains it.  Hopefully an easy fix!

**Reply by pmolodo** [reply]
What's the argument for having distributed builds be in debug mode (even if not for tagged releases)?

**Reply by ErikBjare** [reply]
@pmolodo Much faster build times, but if I don't have a quick fix (tricky it seems) might just have to bite the bullet, for now.

**Reply by pmolodo** [reply]
I see this was addressed [here](https://github.com/ActivityWatch/activitywatch/commit/67bd879979917473754f29a7cad4f5fffc629ae5). Thank you!
